### PR TITLE
v1.2.0 - Implement Bulk versions of APIs and reorganize test project.

### PR DESCRIPTION
**Pull request by Pmac627**

### CODE REVIEW

File: dii.cosmos.tests/DeleteAdapterApiTests.cs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    public class DeleteAdapterApiTests : IClassFixture<AdapterFixture>
+    {
+        private readonly AdapterFixture _adapterFixture;
+
+        public DeleteAdapterApiTests(AdapterFixture adapterFixture)
+        {
+            _adapterFixture = adapterFixture ?? throw new ArgumentNullException(nameof(adapterFixture));
+        }
+
+		#region DeleteAsync
```
Comment (aquamoogle) [inline dii.cosmos.tests/DeleteAdapterApiTests.cs:27]: Goofy indenting

File: dii.cosmos.tests/DeleteAdapterApiTests.cs [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+			_adapterFixture.CreatedFakeEntityTwos.Add(savedFakeEntityTwo);
+		}
+
+		[Fact, TestPriorityOrder(101)]
+		public async Task DeleteAsync_Success()
+		{
+			var fakeEntityTwo = _adapterFixture.CreatedFakeEntityTwos[0];
+
+			var success = await _adapterFixture.FakeEntityTwoAdapter.DeleteAsync(fakeEntityTwo.Id, new PartitionKey(fakeEntityTwo.FakeEntityTwoId)).ConfigureAwait(false);
```
Comment (aquamoogle) [inline dii.cosmos.tests/DeleteAdapterApiTests.cs:63]: PK logic should be inside the adapter.

**Review by aquamoogle — CHANGES_REQUESTED** [review]

File: dii.cosmos.tests/CreateAdapterApiTests.cs [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    public class CreateAdapterApiTests : IClassFixture<AdapterFixture>
+    {
+        private readonly AdapterFixture _adapterFixture;
+
+        public CreateAdapterApiTests(AdapterFixture adapterFixture)
+        {
+            _adapterFixture = adapterFixture ?? throw new ArgumentNullException(nameof(adapterFixture));
+        }
+
+		#region CreateAsync
```
[the remark this answers is not in the document]
Comment (Pmac627) [reply dii.cosmos.tests/CreateAdapterApiTests.cs:27]: Weird anomaly in the PR. Indentation is fine (discussed it).

File: dii.cosmos.tests/DeleteAdapterApiTests.cs (hunk 1 above)
Comment (Pmac627) [reply dii.cosmos.tests/DeleteAdapterApiTests.cs:27]: Weird anomaly in the PR. Indentation is fine (discussed it).

File: dii.cosmos.tests/CreateAdapterApiTests.cs [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+		{
+			var fakeEntityTwo = new FakeEntityTwo
+			{
+				Id = DateTime.Now.Ticks.ToString(),
+				FakeEntityTwoId = DateTime.Now.AddMinutes(-1).Ticks.ToString(),
+				SearchableStringValue = $"fakeEntityTwo: {nameof(FakeEntityTwo.SearchableStringValue)}",
+				CompressedStringValue = $"fakeEntityTwo: {nameof(FakeEntityTwo.CompressedStringValue)}"
+			};
+
+			var savedFakeEntityTwo = await _adapterFixture.FakeEntityTwoAdapter.CreateAsync(fakeEntityTwo, new PartitionKey(fakeEntityTwo.FakeEntityTwoId)).ConfigureAwait(false);
```
[the remark this answers is not in the document]
Comment (Pmac627) [reply dii.cosmos.tests/CreateAdapterApiTests.cs]: Moved to inside the Adapter, as discussed. Still not 100% away from that binding to the Microsoft.Azure.Cosmos.PartitionKey object, but closer.

File: dii.cosmos.tests/CreateAdapterApiTests.cs [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+			};
+
+			var entitiesToCreate = new List<(PartitionKey partitionKey, FakeEntityTwo diiCosmosEntity)>
+			{
+				(new PartitionKey(fakeEntityTwo1.FakeEntityTwoId), fakeEntityTwo1),
+				(new PartitionKey(fakeEntityTwo2.FakeEntityTwoId), fakeEntityTwo2),
+				(new PartitionKey(fakeEntityTwo3.FakeEntityTwoId), fakeEntityTwo3)
+			};
+
+			var savedFakeEntityTwos = await _adapterFixture.FakeEntityTwoAdapter.CreateBulkAsync(entitiesToCreate).ConfigureAwait(false);
```
[the remark this answers is not in the document]
Comment (Pmac627) [reply dii.cosmos.tests/CreateAdapterApiTests.cs:120]: Moved to inside the Adapter, as discussed. Still not 100% away from that binding to the Microsoft.Azure.Cosmos.PartitionKey object, but closer.

File: dii.cosmos.tests/DeleteAdapterApiTests.cs (hunk 2 above)
Comment (Pmac627) [reply dii.cosmos.tests/DeleteAdapterApiTests.cs]: Moved to inside the Adapter, as discussed. Still not 100% away from that binding to the Microsoft.Azure.Cosmos.PartitionKey object, but closer.

**Pull request closed without merging**